bundles/org.openhab.core/src/main/java/org/openhab/core/service/WatchQueueReader.java

```diff
@@ -272,11 +272,9 @@ public void run() {
                 }
 
                 key.reset();
+            } catch (Exception exc) {
+                logger.error("Exception caught in WatchQueueReader: ", exc);
```

Comment:
It is probably better to move the `catch (InterruptedException exc)` for the `key = watchService.take();`  just above this catch. So `InterruptedException`s are still properly handled when they occur when the application is stopping.

If it can recover from these exceptions I don't think it is a good idea to bother end users with this by logging scary stacktraces. When exception handling works as expected there is nothing for them to do and everything will still work.

Furthermore it is always better to use `while (!Thread.currentThread().isInterrupted()) {` instead of `for (;;) {` or `while (true) {` when threads should stop when they are interrupted.

Reply:
> If it can recover from these exceptions I don't think it is a good idea to bother end users with this by logging scary stacktraces. When exception handling works as expected there is nothing for them to do and everything will still work.

I agree but perhaps do a `logger.debug`?

Reply:
Yes it's always good to have some debug logging in case the exception handling needs to be tweaked. Debug logging can also be made very scary. :wink:

Reply:
Can you address this comment so we can merge this PR @ccutrer?

Reply:
I think I've addressed all the comments now - use a conditional loop instead of infinite, move the InterruptedException handling to the outermost loop, change that logging to debug, and try to make the message as friendly as possible even when it is logged.